### PR TITLE
allow custom signal to capture process on event end

### PR DESCRIPTION
**Pull request by JanKoppe**
This patch allows users to configure a custom signal that is sent to the capture process after a configurable time. See #154

### CODE REVIEW

File: pyca/capture.py [hunk 1]
```diff
@@ -115,12 +117,16 @@ def recording_command(event):
 
     # Check process
     while captureproc.poll() is None:
+        if sigcustom_time and timestamp() > sigcustom_time:
```
Comment (lkiesow) [inline pyca/capture.py:120]: No strong opinion on this (basically looks fine like this) but I'm wondering if there is a use-case to ever sending a custom signal followed by a sigterm or if we should merge both and just set the custom signal to sigterm by default. Again, just wondering. In the end, both would be fine with me.

**Comment by JanKoppe** [comment]
I'm not reaally sure about a specific use-case for both a custom `SIG` and a `SIGTERM`. But IMO it doesn't hurt to have the flexibility. If you want me to, I can remove the `SIGTERM` portion, effectively making the old `SIGTERM` changeable.

By the way: This helped us a lot with a stubborn Sony Camera, that sends completely broken streams which ffmpeg will sort of choke on. Using `-t seconds` was impossible to define when the stream ends, and `SIGTERM` regularly destroyed recorded files, but with `SIGINT` it does work very well now. We're just leaving the `-t seconds` portion out of our script entirely now.

File: pyca/capture.py (hunk 1 above)
Comment (JanKoppe) [reply pyca/capture.py:120]: I slightly lean to keeping both - there might be some unforeseen usecase for someone, in which case it's neat to have the flexibility. And I don't think that these lines are problematic to maintain, or a performance-issue, so I see no problem in keeping them.

**Comment by lkiesow** [comment]
Fine with me. I'll try to test this later today.

**Comment by lkiesow** [comment]
Works as expected → merged

**Pull request merged**